File: C#/FrameworksPerformance/Common/Facades/RelationalFrameworkTestsFacade.cs
using Common.Managers;
using Common.Utilites;

namespace Common.Facades
{
    public class RelationalFrameworkTestsFacade : IRelationalFrameworkTestsFacade
    {
        private readonly IRelationalFrameworkManager _manager;
        private readonly JsonManager _jsonManager;

        public RelationalFrameworkTestsFacade(IRelationalFrameworkManager manager, JsonManager jsonManager)
        {
            _manager = manager ?? throw new ArgumentNullException(nameof(manager));
            _jsonManager = jsonManager ?? throw new ArgumentNullException(nameof(jsonManager));
        }

        public string RunAllTests(int samplesQuantity, string fileName)
        {
            var testsResults = new TestsResults();
            testsResults.Add(_manager.SingleRecordSearch(samplesQuantity));
            testsResults.Add(_manager.SetOfDataSearch(samplesQuantity));
            testsResults.Add(_manager.SetOfDataWithIsNullSearch(samplesQuantity));
            testsResults.Add(_manager.AddRecords(samplesQuantity));
            testsResults.Add(_manager.EditRecords(samplesQuantity));
            testsResults.Add(_manager.DeleteRecords(samplesQuantity));
            testsResults.Add(_manager.SearchTwoRelatedTables(samplesQuantity));
            testsResults.Add(_manager.SearchThreeRelatedTables(samplesQuantity));
            testsResults.Add(_manager.SearchRecordsWhichDoesNotHaveConnection(samplesQuantity));
            testsResults.Add(_manager.SearchWithSubQuery(samplesQuantity));

            _jsonManager.SaveTestResultAsFile(fileName,testsResults);
            return _jsonManager.ConvertToJson(testsResults);
        }

        public string SingleRecordSearch(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.SingleRecordSearch(samplesQuantity));

        public string SetOfDataSearch(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.SetOfDataSearch(samplesQuantity));

        public string SetOfDataWithIsNullSearch(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.SetOfDataWithIsNullSearch(samplesQuantity));

        public string AddRecords(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.AddRecords(samplesQuantity));

        public string EditRecords(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.EditRecords(samplesQuantity));

        public string DeleteRecords(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.DeleteRecords(samplesQuantity));

        public string SearchTwoRelatedTables(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.SearchTwoRelatedTables(samplesQuantity));

        public string SearchFourRelatedTables(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.SearchThreeRelatedTables(samplesQuantity));

        public string SearchRecordsWhichDoesNotHaveConnection(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.SearchRecordsWhichDoesNotHaveConnection(samplesQuantity));

        public string SearchWithSubQuery(int samplesQuantity) => _jsonManager.ConvertToJson(_manager.SearchWithSubQuery(samplesQuantity));
    }
}

File: Java-sqlite/JavaDatabaseFrameworks/Common/src/main/java/org/dbFrameworks/Facade/RelationalFrameworkTestsFacade.java
package org.dbFrameworks.Facade;

import org.dbFrameworks.Managers.IRelationalFrameworkManager;
import org.dbFrameworks.TestsResults;

public class RelationalFrameworkTestsFacade implements IRelationalFrameworkTestsFacade {
    IRelationalFrameworkManager _relationalFrameworkManager;

    public RelationalFrameworkTestsFacade(IRelationalFrameworkManager relationalFrameworkManager) {
        _relationalFrameworkManager = relationalFrameworkManager;
    }

    @Override
    public String RunAllTests(int samplesQuantity, String fileName) {
        var testsResults = new TestsResults();
        testsResults.Add(_relationalFrameworkManager.SingleRecordSearch(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.SetOfDataSearch(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.SetOfDataWithIsNullSearch(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.AddRecords(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.EditRecords(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.DeleteRecords(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.SearchTwoRelatedTables(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.SearchFourRelatedTables(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.SearchRecordsWhichDoesNotHaveConnection(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.SearchWithSubQuery(samplesQuantity));
        testsResults.Add(_relationalFrameworkManager.RemoveRelatedRecords(samplesQuantity));
        return testsResults.toString();
    }

    @Override
    public String SingleRecordSearch(int samplesQuantity) {
        return _relationalFrameworkManager.SingleRecordSearch(samplesQuantity).toString();
    }

    @Override
    public String SetOfDataSearch(int samplesQuantity) {
        return _relationalFrameworkManager.SetOfDataSearch(samplesQuantity).toString();
    }

    @Override
    public String SetOfDataWithIsNullSearch(int samplesQuantity) {
        return _relationalFrameworkManager.SetOfDataWithIsNullSearch(samplesQuantity).toString();
    }

    @Override
    public String AddRecords(int samplesQuantity) {
        return _relationalFrameworkManager.AddRecords(samplesQuantity).toString();
    }

    @Override
    public String EditRecords(int samplesQuantity) {
        return _relationalFrameworkManager.EditRecords(samplesQuantity).toString();
    }

    @Override
    public String DeleteRecords(int samplesQuantity) {
        return _relationalFrameworkManager.DeleteRecords(samplesQuantity).toString();
    }

    @Override
    public String SearchTwoRelatedTables(int samplesQuantity) {
        return _relationalFrameworkManager.SearchTwoRelatedTables(samplesQuantity).toString();
    }

    @Override
    public String SearchFourRelatedTables(int samplesQuantity) {
        return _relationalFrameworkManager.SearchFourRelatedTables(samplesQuantity).toString();
    }

    @Override
    public String SearchRecordsWhichDoesNotHaveConnection(int samplesQuantity) {
        return _relationalFrameworkManager.SearchRecordsWhichDoesNotHaveConnection(samplesQuantity).toString();
    }

    @Override
    public String SearchWithSubQuery(int samplesQuantity) {
        return _relationalFrameworkManager.SearchWithSubQuery(samplesQuantity).toString();
    }

    @Override
    public String RemoveRelatedRecords(int samplesQuantity) {
        return _relationalFrameworkManager.RemoveRelatedRecords(samplesQuantity).toString();
    }

}
